plugins/flytekit-polars/flytekitplugins/polars/schema.py

```diff
@@ -0,0 +1,121 @@
+        raise AssertionError("Only Parquet type files are supported for polars dataframe currently")
+
+
+class PolarsSchemaWriter(LocalIOSchemaWriter[pl.DataFrame]):
+    """
+    Implements how polars.DataFrame should be written to using ``open`` method of FlyteSchema
+    """
+
+    def __init__(
+        self,
+        to_path: os.PathLike,
+        cols: typing.Optional[typing.Dict[str, type]],
+        fmt: SchemaFormat,
+    ):
+        super().__init__(str(to_path), cols, fmt)
+
+    def write(self, *dfs: pl.DataFrame, **kwargs):
+        if dfs is None or len(dfs) == 0:
+            return
+        if len(dfs) > 1:
+            raise AssertionError("Only a single polars.DataFrame can be written per variable currently")
+        if self._fmt == SchemaFormat.PARQUET:
+            out_path = self.to_path + "/00000"
+            df = dfs[0]
+
+            # Polars 0.13.12 deprecated to_parquet in favor of write_parquet
+            if hasattr(df, "write_parquet"):
+                df.write_parquet(out_path)
+            else:
+                df.to_parquet(out_path)
+            return
+        raise AssertionError("Only Parquet type files are supported for polars dataframe currently")
+
+
+class PolarsDataFrameTransformer(TypeTransformer[pl.DataFrame]):
+    """
+    Transforms polars.DataFrame to and from a Schema (typed/untyped)
+    """
+
+    _SUPPORTED_TYPES: typing.Dict[
+        type, SchemaType.SchemaColumn.SchemaColumnType
+    ] = FlyteSchemaTransformer._SUPPORTED_TYPES
+
+    def __init__(self):
+        super().__init__("polars-df-transformer", pl.DataFrame)
+
+    @staticmethod
+    def _get_schema_type() -> SchemaType:
+        return SchemaType(columns=[])
+
+    def get_literal_type(self, t: Type[pl.DataFrame]) -> LiteralType:
+        return LiteralType(schema=self._get_schema_type())
+
+    def to_literal(
+        self,
+        ctx: FlyteContext,
+        python_val: pl.DataFrame,
+        python_type: Type[pl.DataFrame],
+        expected: LiteralType,
+    ) -> Literal:
+        local_dir = ctx.file_access.get_random_local_directory()
+        w = PolarsSchemaWriter(to_path=local_dir, cols=None, fmt=SchemaFormat.PARQUET)
+        w.write(python_val)
+        remote_path = ctx.file_access.get_random_remote_directory()
+        ctx.file_access.put_data(local_dir, remote_path, is_multipart=True)
+        return Literal(scalar=Scalar(schema=Schema(remote_path, self._get_schema_type())))
+
+    def to_python_value(
+        self,
+        ctx: FlyteContext,
+        lv: Literal,
+        expected_python_type: Type[pl.DataFrame],
+    ) -> T:
+        if not (lv and lv.scalar and lv.scalar.schema):
+            return pl.DataFrame()
+        local_dir = ctx.file_access.get_random_local_directory()
+        ctx.file_access.download_directory(lv.scalar.schema.uri, local_dir)
+        r = PolarsSchemaReader(from_path=local_dir, cols=None, fmt=SchemaFormat.PARQUET)
+        return r.all()
+
+
+SchemaEngine.register_handler(

```

Comment:
@RobinKa Thank you for your work. We may deprecate the scheme engine in the future. could you register this handler to `StructuredDatasetTransformerEngine` instead? Check this example https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-spark/flytekitplugins/spark/sd_transformers.py

Reply:
Do we still want the schema stuff at all or do we only want the StructuredDataset ones?

Reply:
we only want the StructuredDataset ones